app/src/main/java/com/framgia/fsalon/screen/billdialog/BillDialogViewModel.java

```diff
@@ -0,0 +1,221 @@
+package com.framgia.fsalon.screen.billdialog;
+
+import android.databinding.BaseObservable;
+import android.databinding.Bindable;
+import android.support.v4.app.DialogFragment;
+import android.support.v4.app.Fragment;
+import android.widget.ArrayAdapter;
+
+import com.framgia.fsalon.BR;
+import com.framgia.fsalon.data.model.BillItemRequest;
+import com.framgia.fsalon.data.model.Service;
+import com.framgia.fsalon.data.model.Stylist;
+import com.framgia.fsalon.screen.bill.BillAdapter;
+import com.framgia.fsalon.screen.bill.BillViewModel;
+
+/**
+ * Exposes the data to be used in the BillDialog screen.
+ */
+public class BillDialogViewModel extends BaseObservable implements BillDialogContract.ViewModel {
+    private BillDialogContract.Presenter mPresenter;
+    private ArrayAdapter<Stylist> mStylistAdapter;
+    private ArrayAdapter<Service> mServiceAdapter;
+    private BillAdapter mAdapter;
+    private Stylist mStylist;
+    private Service mService;
+    private String mPrice;
+    private String mQty;
+    private float mTotal;
+    private BillItemRequest mBillItemRequest;
+    private int stylistPos;
+    private int servicePos;
+    private Fragment mFragment;
+    private BillViewModel mViewModelBill;
+
+    public BillDialogViewModel(ArrayAdapter<Stylist> stylistAdapter,
+                               ArrayAdapter<Service> serviceAdapter,
+                               BillItemRequest itemRequest,
+                               Fragment fragment,
+                               BillViewModel billViewModel
+    ) {
+        mStylistAdapter = stylistAdapter;
+        mServiceAdapter = serviceAdapter;
+        mBillItemRequest = itemRequest;
+        mPrice = String.valueOf(itemRequest.getPrice());
+        mQty = String.valueOf(itemRequest.getQty());
+        mTotal = itemRequest.getRowTotal();
+        mFragment = fragment;
+        mViewModelBill = billViewModel;
+    }
+
+    @Override
+    public void onStart() {
+        mPresenter.onStart();
+    }
+
+    @Override
+    public void onStop() {
+        mPresenter.onStop();
+    }
+
+    public void onCancelClick() {
+        ((DialogFragment) mFragment).dismiss();
+    }
+
+    public void onUpdateClick() {
+        BillItemRequest bill = new BillItemRequest.Builder()
+            .setServiceProductId(mService.getId())
+            .setStylistId(mStylist.getId())
+            .setPrice(Float.valueOf(mPrice))
+            .setQty(Integer.valueOf(mQty))
```

Comment:
ERROR: [PERFORMANCE][DM_BOXED_PRIMITIVE_FOR_PARSING] Boxing/unboxing to parse a primitive com.framgia.fsalon.screen.billdialog.BillDialogViewModel.onUpdateClick()
http://findbugs.sourceforge.net/bugDescriptions.html#DM_BOXED_PRIMITIVE_FOR_PARSING